velox/exec/tests/PlanNodeSerdeTest.cpp

```diff
@@ -0,0 +1,292 @@
+          makeFlatVector<int64_t>({10, 20, 30}),
+          makeFlatVector<bool>({true, true, false}),
+      });
+
+  auto build = makeRowVector(
+      {"u0", "u1", "u2"},
+      {
+          makeFlatVector<int32_t>({1, 2, 3}),
+          makeFlatVector<int64_t>({10, 20, 30}),
+          makeFlatVector<bool>({true, true, false}),
+      });
+
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  auto plan =
+      PlanBuilder(planNodeIdGenerator)
+          .values({probe})
+          .hashJoin(
+              {"t0"},
+              {"u0"},
+              PlanBuilder(planNodeIdGenerator).values({build}).planNode(),
+              "t1 > u1",
+              {"t0", "t1", "u2", "t2"},
+              core::JoinType::kInner)
+          .planNode();
+
+  testSerde(plan);
+
+  plan = PlanBuilder(planNodeIdGenerator)
+             .values({probe})
+             .hashJoin(
+                 {"t0"},
+                 {"u0"},
+                 PlanBuilder(planNodeIdGenerator).values({build}).planNode(),
+                 "", // no filter
+                 {"t0", "t1", "u2", "t2"},
+                 core::JoinType::kLeft)
+             .planNode();
+
+  testSerde(plan);
+}
+
+TEST_F(PlanNodeSerdeTest, topN) {
+  auto plan = PlanBuilder().values({data_}).topN({"c0"}, 10, true).planNode();
+  testSerde(plan);
+
+  plan = PlanBuilder().values({data_}).topN({"c0 DESC"}, 10, false).planNode();
+  testSerde(plan);
+
+  plan = PlanBuilder()
+             .values({data_})
+             .topN({"c0", "c1 DESC NULLS FIRST"}, 10, true)
+             .planNode();
+  testSerde(plan);
+}
+
+TEST_F(PlanNodeSerdeTest, unnest) {
+  auto data = makeRowVector({
+      makeFlatVector<int64_t>({1, 2, 3}),
+      makeArrayVector<int32_t>({
+          {1, 2},
+          {3, 4, 5},
+          {},
+      }),
+      makeMapVector<int64_t, int32_t>(
+          {{{1, 10}, {2, 20}}, {{3, 30}, {4, 40}, {5, 50}}, {}}),
+  });
+
+  auto plan = PlanBuilder()
+                  .values({data})
+                  .unnest({"c0"}, {"c1", "c2"}, std::nullopt)
+                  .planNode();
+  testSerde(plan);
+
+  plan =
+      PlanBuilder().values({data}).unnest({"c0"}, {"c1"}, "ordinal").planNode();
+  testSerde(plan);
+}
+
+TEST_F(PlanNodeSerdeTest, values) {
+  auto plan = PlanBuilder().values({data_}).planNode();
+  testSerde(plan);
+
+  plan = PlanBuilder()
+             .values({data_}, true /*parallelizable*/, 5 /*repeatTimes*/)
+             .planNode();
+  testSerde(plan);
+}
+
+TEST_F(PlanNodeSerdeTest, window) {
+  auto plan =
+      PlanBuilder()
+          .values({data_})
+          .window(
+              {"sum(c0) over (partition by c1 order by c2 rows between 10 preceding and 5 following)"})
+          .planNode();
+
+  testSerde(plan);
+
+  plan = PlanBuilder()
+             .values({data_})
+             .window({"sum(c0) over (partition by c1 order by c2)"})
+             .planNode();
+
+  testSerde(plan);
+}
```

Comment:
nit: another small good practice is to add a test suite expecting the ones not yet implemented to fail. So that when people do implement them, this test fails and they are reminded to add tests to cover the newly added code.

Reply:
I'm not sure about this. I expect folks adding new functionality to always add a test and for reviewers to remind about this as needed. Adding test cases that fail will make this PR even larger and I will either have to think about all the scenarios that need testing or add just one quick test case for each plan node. If I do the latter, the person implementing the functionality may feel they just need to "fix" that single test case and not add all the necessary test cases.